jerry-core/ecma/base/ecma-gc.c

```diff
@@ -347,6 +347,46 @@ ecma_gc_mark (ecma_object_t *object_p) /**< object to mark from */
 } /* ecma_gc_mark */
 
 /**
+ * Free the native handle/pointer by calling its free callback
+ */
+static void
+ecma_gc_free_native (ecma_property_t *property_p, /**< property */
+                     lit_magic_string_id_t id) /**< identifier of internal property */
+{
+  JERRY_ASSERT (property_p != NULL);
+
+  JERRY_ASSERT (id == LIT_INTERNAL_MAGIC_STRING_NATIVE_HANDLE
+                || id == LIT_INTERNAL_MAGIC_STRING_NATIVE_POINTER);
+
+  ecma_property_value_t *value_p = ECMA_PROPERTY_VALUE_PTR (property_p);
+  ecma_external_pointer_t native_p;
+  ecma_external_pointer_t free_cb;
+  void *package_p;
+
+#ifdef ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY
+  package_p = value_p->value;
+#else /* !ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY */
+  package_p = ECMA_GET_NON_NULL_POINTER (void, value_p->value);
+#endif /* ECMA_VALUE_CAN_STORE_UINTPTR_VALUE_DIRECTLY */
```

Comment:
There are two macros for this: ECMA_GET/SET_INTERNAL_VALUE_POINTER, please use them.